Clade.js

```diff
@@ -0,0 +1,682 @@
+
+		// Extra
+		getExtra(record, newItem);
+
+		// Tagging
+		getTags(record, newItem);
+
+		// Doc type
+		var itemType = getDocumentType(doc);
+		if (itemType) {
+			newItem.itemType = itemType;
+		}
+
+		// Repository
+		newItem.libraryCatalog = "Clade (https://bibliotheques-numeriques.defense.gouv.fr)";
+	}
+
+	// Check for Gallica URL (digital version available), if found, set item.url
+	function checkGallica(record, item) {
+		var url = record.getFieldSubfields("856");
+
+		if (url && url.length > 0 && url[0].u) {
+			item.url = url[0].u;
+		}
+	}
+
+	this.marcURL = function (url) {
+		return url + '/marcxml';
+	};
+
+	this.processMarcUrl = function (doc, marcurl) {
+		Zotero.Utilities.HTTP.doGet(marcurl, function (xmlText) {
+			var translator = Zotero.loadTranslator("import");
+
+			// Use MARC translator
+			translator.setTranslator("a6ee60df-1ddc-4aae-bb25-45e0537be973");
+			translator.getTranslatorObject(function (marc) {
+				Clade.transformMarcXml(doc, marc, xmlText);
+			});
+			translator.translate();
+		}); // doGet end
+	};
+
+	// Transform MarcXML to Marc base on translator MARCXML (translatorID: edd87d07-9194-42f8-b2ad-997c4c7deefd)
+	this.transformMarcXml = function (doc, marc, xmlText) {
+		var parser = new DOMParser();
+		var xml = parser.parseFromString(xmlText, 'text/xml');
+		// define the marc namespace
+		var ns = {
+			marc: "http://www.loc.gov/MARC21/slim"
+		};
+
+		var records = ZU.xpath(xml, '//marc:record', ns);
+		for (let rec of records) {
+			// create one new item per record
+			var record = new marc.record();
+			var newItem = new Zotero.Item();
+			record.leader = ZU.xpathText(rec, "./marc:leader", ns);
+			var fields = ZU.xpath(rec, "./marc:datafield", ns);
+			for (let field of fields) {
+				// go through every datafield (corresponds to a MARC field)
+				var subfields = ZU.xpath(field, "./marc:subfield", ns);
+				var tag = "";
+				for (let subfield of subfields) {
+					// get the subfields and their codes...
+					var code = ZU.xpathText(subfield, "./@code", ns);
+					var sf = ZU.xpathText(subfield, "./text()", ns);
+					// delete non-sorting symbols
+					// e.g. &#152;Das&#156; Adam-Smith-Projekt
+					if (sf) {
+						sf = sf.replace(/[\x80-\x9F]/g, "");
+						// concat all subfields in one datafield, with subfield delimiter and code between them
+						tag = tag + marc.subfieldDelimiter + code + sf;
+					}
+				}
+				record.addField(ZU.xpathText(field, "./@tag", ns), ZU.xpathText(field, "./@ind1", ns) + ZU.xpathText(field, "./@ind2"), tag);
+			}
+
+			record.translate(newItem);
+
+			// Do specific Unimarc postprocessing
+			postProcessMarc(doc, record, newItem);
+
+			// Check for Gallica URL
+			checkGallica(record, newItem);
+
+			newItem.complete();
+		}
+	};
+};
+
+/* Global CladeClass object. */
+var Clade = new CladeClass();
+
+function detectWeb(doc, url) {
+	if (url.search(/\/document\//) != -1) {
+		return getDocumentType(doc);
+	}
+
+	return undefined;
```

Comment:
This script is look like BnF translator because on project the customer would same display on Zotero than BnF but we haven't same html tag or url pattern.

The MARC translator doesn't work because a final user have html page on doesn't show xml so Zotero doesn't use this translator. And default MARCXML translator is incomplete for usage would by customer and missing some processing on fields.